apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx

```diff
@@ -428,6 +431,43 @@ export function WriteInsAdjudicationScreen(): JSX.Element {
     ...writeInCandidates.map((c) => normalizeWriteInName(c.name)),
   ];
 
+  const writeInImageView = writeInImageViewQuery.data;
+  const writeInAdjudicationContext = writeInAdjudicationContextQuery.data;
+  const currentWriteIn = writeInAdjudicationContext?.writeIn;
+  const currentWriteInMarkedInvalid =
+    currentWriteIn &&
+    currentWriteIn.status === 'adjudicated' &&
+    currentWriteIn.adjudicationType === 'invalid';
+
+  // these IDs cannot be selected because the voter filled in a bubble for the candidate
+  const markedOfficialCandidateIds = writeInAdjudicationContext
+    ? (
+        (writeInAdjudicationContext.cvrVotes[contestId] as CandidateId[]) ?? []
+      ).filter((candidateId) => !candidateId.startsWith('write-in-'))
+    : [];
+  // these IDs cannot be selected because another write-in on this ballot
+  // has already been adjudicated for the same official candidate
+  const writeInAdjudicatedOfficialCandidateIds = writeInAdjudicationContext
+    ? writeInAdjudicationContext.relatedWriteIns
+        .filter(
+          (w): w is WriteInRecordAdjudicatedOfficialCandidate =>
+            w.status === 'adjudicated' &&
+            w.adjudicationType === 'official-candidate'
+        )
+        .map((w) => w.candidateId)
+    : [];
+  // these IDs cannot be selected because another write-in on this ballot
+  // has already been adjudicated for the same write-in candidate
+  const writeInAdjudicatedWriteInCandidateIds = writeInAdjudicationContext
+    ? writeInAdjudicationContext.relatedWriteIns
+        .filter(
+          (w): w is WriteInRecordAdjudicatedOfficialCandidate =>
+            w.status === 'adjudicated' &&
+            w.adjudicationType === 'write-in-candidate'
+        )
+        .map((w) => w.candidateId)
+    : [];
+
```

Comment:
This logic of identifying off-limits adjudication options was originally in the backend, but it made more sense to just give the frontend all the data and figure out the limits on the frontend.